### PR TITLE
Add new host for running single-exe bundle prototype

### PR DESCRIPTION
**Pull request by elinor-fung**
Initial addition of a basic host for running as a bundle. Only on Linux. Based on unixcoreconsole.

Everything under the `bundle/` subdirectory is basically a copy of the files from core-setup (as of 4b82b1e). I also copied pal/utils from core-setup and removed the parts we weren't using. There's probably more that could be pared down, but I wanted to get a starting point in for the prototype.

This host can be copied over the real 'apphost' in an SDK install such that it will be used to create the bundle instead.

### CODE REVIEW

**Comment by elinor-fung** [comment]
@swaroop-sridhar I just copied over the bundle code as it exists now - i.e. just extracting files, not including the memory mapping changes you have in flight.

File: src/coreclr/hosts/unixcorebundle/corebundle.cpp [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    {
+        perror("Could not extract contents of the bundle");
+        return bundle_status;
+    }
+
+    std::string root_dir = extractor.get_extraction_dir();
+    std::string app_path(root_dir);
+    app_path.push_back(DIR_SEPARATOR);
+
+    std::string binding(&embed[0]);
```
Comment (swaroop-sridhar) [inline src/coreclr/hosts/unixcorebundle/corebundle.cpp]: For simplicity in the prototype, we could assume the name of the app to be <hostname>.dll. 
The change is fine as-is too, we'll need to external tooling to update the blob using HostModel utilities.

**Comment by swaroop-sridhar** [comment]
> @swaroop-sridhar I just copied over the bundle code as it exists now - i.e. just extracting files, not including the memory mapping changes you have in flight.

This is fine, I think we can replace the `\bundle` directory en masse once the changes are checked in to core-setup.

**Comment by swaroop-sridhar** [comment]
@dotnet-bot  test this please.

**Comment by elinor-fung** [comment]
@swaroop-sridhar yeah, beyond the prototype, we'd need some sort of json parsing (runtime configs, deps). I'm hoping we may be able to reasonably estimate what added overhead that would bring based on measurements of hostpolicy.

What is expected to be the big cost for startup hooks on the host? I thought that was just passing in another runtime property based on an environment variable?

**Comment by swaroop-sridhar** [comment]
>> I'm hoping we may be able to reasonably estimate what added overhead that would bring based on measurements of hostpolicy.

OK, please share the number when you get to the measurement. `hostpolicy.dll` is about 0.5 MB -- which is quite steep to pay in the scenario. I'm not sure how much of it (wrt code-size wise) will carry over.

>> What is expected to be the big cost for startup hooks on the host? 

I agree about startup-hook; I mainly wrote it down as an example of things we've removed but will be present in the full set (to keep the size calculations as good as possible).

**Comment by elinor-fung** [comment]
Tried just ripping out the json parsing and deps resolving from hostpolicy - looked like a difference of ~160KB (Linux x64). From Leandro's investigations into switching to RapidJSON that was ~79KB less than casablanca. So in terms of size, I think a very rough estimate would be +80KB (with the switch to RapidJSON).

**Comment by swaroop-sridhar** [comment]
Thanks for the explanation @elinor-fung 

.

**Pull request merged**